tests/src/test/java/org/wildfly/plugins/bootablejar/maven/goals/AbstractBootableJarMojoTestCase.java

```diff
@@ -0,0 +1,364 @@
+        checkURL(dir, null, createUrl(TestEnvironment.getHttpPort(), isRoot ? "" : "test"), true, args);
+    }
+
+    protected void checkManagementItf(Path dir, boolean start) throws Exception {
+        checkURL(dir, null, createUrl(TestEnvironment.getManagementPort(), "management"), start);
+    }
+
+    protected void checkMetrics(Path dir, boolean start) throws Exception {
+        checkURL(dir, null, createUrl(TestEnvironment.getManagementPort(), "metrics"), start);
+    }
+
+    protected void checkURL(Path dir, String fileName, String url, boolean start, String... args) throws Exception {
+        Process process = null;
+        int timeout = TestEnvironment.getTimeout() * 1000;
+        long sleep = 1000;
+        boolean success = false;
+        try {
+            if (start) {
+                process = startServer(dir, fileName, args);
+                try (
+                        ModelControllerClient client = ModelControllerClient.Factory.create(TestEnvironment.getHost(),
+                                TestEnvironment.getManagementPort())
+                ) {
+                    // Wait for the server to start
+                    ServerHelper.waitForStandalone(process, client, TestEnvironment.getTimeout());
+                }
+            }
+            while (timeout > 0) {
+                if (checkURL(url)) {
+                    System.out.println("Successfully connected to " + url);
+                    success = true;
+                    break;
+                }
+                Thread.sleep(sleep);
+                timeout -= sleep;
+            }
+            shutdownServer(dir);
+            // If the process is not null wait for it to shutdown
+            if (process != null) {
+                assertTrue("The process has failed to shutdown", process.waitFor(TestEnvironment.getTimeout(), TimeUnit.SECONDS));
+            }
+        } finally {
+            ProcessHelper.destroyProcess(process);
+        }
+        if (!success) {
+            throw new Exception("Unable to interact with deployed application");
+        }
+    }
+
+    protected Process startServer(Path dir, String fileName, String... args) throws Exception {
+        List<String> cmd = new ArrayList<>();
+        cmd.add(getJavaCommand());
+        cmd.addAll(getJvmArgs());
+        cmd.add("-jar");
+        cmd.add(dir.resolve("target").resolve(fileName == null ? TEST_FILE : fileName).toAbsolutePath().toString());
+        cmd.addAll(Arrays.asList(args));
+        final Path out = TestEnvironment.createTempPath("logs", getClass().getName() + "-process.txt");
+        final Path parent = out.getParent();
+        if (parent != null && Files.notExists(parent)) {
+            Files.createDirectories(parent);
+        }
+        return new ProcessBuilder(cmd)
+                .redirectErrorStream(true)
+                .redirectOutput(out.toFile())
+                .start();
+        // We can't use start in tests, breaks maven test execution.
```

Comment:
Is this comment (and following block) still needed? Did it come out after the refactoring?

Reply:
I just left it because it was there in the original test. I'm not really sure if we need it, but I didn't want to change too much of the actual code.